### PR TITLE
[Scripts] Update MacOS script to C++17

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

[Update MacOS script to C++17](https://github.com/KratosMultiphysics/Kratos/commit/1d93be6b6d850277b0f483d28ed55136e00264fb). Also updating instruction set and python version

**🆕 Changelog**

- [Update MacOS script to C++17](https://github.com/KratosMultiphysics/Kratos/commit/1d93be6b6d850277b0f483d28ed55136e00264fb)

### CODE REVIEW

File: scripts/standard_configure_mac.sh [hunk 1]
```diff
@@ -43,11 +43,13 @@ rm -rf "${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}/CMakeCache.txt"
...
 # Configure
-/Applications/CMake.app/Contents/bin/cmake -H"${KRATOS_SOURCE}" -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}"    \
- -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -msse3 -std=c++11 -L/usr/local/opt/llvm/lib"                         \
- -DCMAKE_C_FLAGS="${CMAKE_C_FLAGS} -msse3 -L/usr/local/opt/llvm/lib"                                        \
- -DUSE_EIGEN_MKL=OFF                                                                                        \
- -DKRATOS_GENERATE_PYTHON_STUBS=ON                                                                          \
+/Applications/CMake.app/Contents/bin/cmake                                                 \
+ -H"${KRATOS_SOURCE}"                                                                      \
+ -B"${KRATOS_BUILD}/${KRATOS_BUILD_TYPE}"                                                  \
+ -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -march=native -std=c++17 -L/usr/local/opt/llvm/lib" \
```
Comment (philbucher) [inline scripts/standard_configure_mac.sh]:
```suggestion
 -DCMAKE_CXX_FLAGS="${CMAKE_CXX_FLAGS} -march=native -L/usr/local/opt/llvm/lib" \
```

I would remove this unless it is absolutely necessary and let CMake handle this

**Comment by roigcarlo** [comment]
Btw, do someone has the means to test this? I am not acquainted with mac compilation with recent versions of macOS. Is very possible that this script does not work anymore.

**Comment by loumalouomega** [comment]
> Btw, do someone has the means to test this? I am not acquainted with mac compilation with recent versions of macOS. Is very possible that this script does not work anymore.

Any MacOS user active on this?

**Comment by loumalouomega** [comment]
> @roigcarlo do you know why the CI was triggered here?

It is not part of the exceptions?

File: scripts/standard_configure_mac.sh (hunk 1 above)
Comment (loumalouomega) [reply scripts/standard_configure_mac.sh]: Done

**Comment by roigcarlo** [comment]
It should, I assume the wildcard is not working properly

**Comment by roigcarlo** [comment]
Should be fixed in #10226

**Comment by loumalouomega** [comment]
Can we proceed with this?, or do I close it?

**Comment by roigcarlo** [comment]
If no one has mac to test I will do it myself. but this has been open for less than a day. I have no time to review all the PR's in such short time and tbh this is low priority in the list. 

Leave it open, but expect it to take a while.

**Pull request closed without merging**